pkg/kvstore/allocator/allocator.go

```diff
@@ -438,12 +444,37 @@ func (k *kvstoreBackend) RunGC(ctx context.Context, rateLimit *rate.Limiter, sta
 
 	staleKeys := map[string]uint64{}
 
+	min := uint64(minID)
+	max := uint64(maxID)
+	reasonOutOfRange := fmt.Sprintf("out of local cluster identity range [%d,%d]", min, max)
+
 	// iterate over /id/
 	for key, v := range allocated {
 		// if k.lockless {
 		// FIXME: Add DeleteOnZeroCount support
 		// }
 
+		// Parse identity ID
+		items := strings.Split(key, "/")
+		if len(items) == 0 {
+			log.WithField(fieldKey, key).WithError(err).Warning("Unknown identity key found, skipping")
+			continue
+		}
+
+		if identityID, err := strconv.ParseUint(items[len(items)-1], 10, 32); err != nil {
```

Comment:
Shouldn't this be `strconv.ParseUint(items[len(items)-1], 10, 64)` instead of `32`?

Reply:
Using `32` instead of `64` considers the fact that we are parsing an identity from its numeric representation, the type of which is [defined as uint32](https://github.com/cilium/cilium/blob/master/pkg/identity/numericidentity.go#L379). Not sure if this make sense?

On the other hand, using `32` or `64` will result to the same parsed value, and there seems no side-effects, so I'm ok to both.

@aanm Thanks!

Reply:
yes it does make sense but I got confused by lines 447 and 448:
```
	min := uint64(minID)
	max := uint64(maxID)
```

so shouldn't those ones be `uint32` in that case?

Reply:
Yes, I think here they can safely be converted to `uint32` from `idpool.ID` (which is `uint64`). But considering `strconv.ParseUint` always returns a `uint64` value, for the comparison operation, we need to convert the return value to `uint32` too.

Reply:
For consistency and brivity, should I just replace `32` with `64`?

Reply:
yes, that's what I was initially suggesting :smile:

Reply:
Agreed, using `64` consistently is the right way to go here.

Reply:
Done and rebased to the latest master, thanks! @aanm @twpayne